### PR TITLE
fire `s-expandable-control:show` and `s-expandable-control:hide` events

### PR DESCRIPTION
**Pull request by balpha**
Addresses #269.

### CODE REVIEW

**Comment by aaronshekey** [comment]
I'd like to see an example of this documented, if possible. For a future PR, I'd like to see a more formal table of the javascript attributes similar to https://stackoverflow.design/product/components/tables#javascript-data-attributes

**Comment by balpha** [comment]
Not sure what you mean -- I didn't add any attributes.

Also not sure what kind of examples you'd like to see.

**Comment by aaronshekey** [comment]
> Not sure what kind of examples you'd like to see.

How about we toggle the class `is-selected` on this page's example? Or we can have another example further down the page:

image

We currently document that the event exists, but we don't answer the question "Ok, but how do I use it?"

> I didn't add any attributes

I think it would be helpful in a future PR to document these in the same way we did with the attributes on sortable tables. I know you didn't add any attributes.

File: lib/js/stacks.js [hunk 1]
```diff
@@ -13,13 +13,25 @@
...
+    };
     StacksController.prototype.setElementData = function(element, key, value) {
         element.setAttribute("data-" + this.identifier + "-" + key, value);
-    }
+    };
     StacksController.prototype.removeElementData = function(element, key) {
         element.removeAttribute("data-" + this.identifier + "-" + key);
-    }
+    };
+    StacksController.prototype.triggerEvent = function(eventName, optionalElement) {
```
Comment (bnickel) [inline lib/js/stacks.js]: It's not strictly needed here, but could we upgrade this to a custom event so we can pass data?

```
    StacksController.prototype.triggerEvent = function(eventName, detail, optionalElement) {
	        var event;
	        var namespacedName = this.identifier + ":" + eventName;
	        try {
	            event = new CustomEvent(namespacedName, {bubbles: true, detail: detail});
	        } catch (ex) {
	            // Internet Explorer
	            event = document.createEvent("CustomEvent");
	            event.initCustomEvent(namespacedName, true, true, detail);
	        }
	        (optionalElement || this.element).dispatchEvent(event);
	    };
```

I have a controller that wants to post a deletion event for another to read:

```
Stacks.addController('foo', {
   removeItem(e) {
    var el = e.currentTarget;
    var id = this.getElementData(el, 'id');
    el.parentNode.removeChild(el);
    this.triggerEvent('removed', { id: id });
  },
});

Stacks.addController('bar', {
   logRemoval(e) {
    var id = e.detail.id;
    this.element.textContent = `${id} removed`;
  },
});
```

```
<div data-controller="foo">
  <button data-foo-id="123" data-action="removeItem">Remove Me</button>
</div>
<div data-controller="bar" data-action="foo:removed@document->bar#logRemoval"></div>
```

Comment (balpha) [reply lib/js/stacks.js]: Makes a lot of sense! Wanna commit that?

**Comment by aaronshekey** [comment]
How's this one doing with Brian's changes, @balpha?

**Comment by balpha** [comment]
Those changes are fine, I asked him to commit them (see above).

**Pull request merged**